File: frontend/src/types/book.ts
export interface BookAuthor {
  id: string;
  firstName: string;
  lastName: string;
  authorOrder: number;
}

export interface BookCategory {
  id: string;
  name: string;
}

export interface Book {
  id: string;
  isbn: string;
  title: string;
  description?: string;
  pageCount: number;
  publicationYear: number;
  language: string;

  publisherId: string;
  publisherName: string;

  formatId: string;
  formatName: string;

  categories: BookCategory[];

  coverImageUrl?: string;

  authors: BookAuthor[];
}

File: backend/Entities/Book.cs
namespace backend.Entities;

public class Book
{
    public Guid Id { get; set; }

    public string ISBN { get; set; } = string.Empty;

    public string Title { get; set; } = string.Empty;

    public string? Description { get; set; }

    public int PageCount { get; set; }

    public short PublicationYear { get; set; }

    public string Language { get; set; } = string.Empty;

    public Guid PublisherId { get; set; }

    public Guid FormatId { get; set; }

    public string? CoverImageUrl { get; set; }

    public DateTimeOffset CreatedAt { get; set; }

    public DateTimeOffset UpdatedAt { get; set; }
}